### DISCUSSION TITLE
Creating a record in a collection that relates to other collections

### DISCUSSION THREAD

**Discussion by steve-potts** [pocketbase/pocketbase#2722] [Q&A] [answered]
Sorry all i have another question,
What is the correct way to add a Customer for example, with related Customer Email. customer number etc.

Here is how i currently have it configured, but Lets say i want to create a new customer and at the same time create a customer email, etc.

![image](https://github.com/pocketbase/pocketbase/assets/4457038/c941fbc0-2dae-454b-99ac-d143ac77e56d)

Have i got this the wrong way around?

I found a similar question on here where someone asked the same kind of question but relating to questions and answers, and they kind of had the relations in reverse to me, so that the answers related to the question, in my example that would be the customers address having a customerId, whereas currently i have it so that a Customer record has a CustomerAddressId(s)

Be good to get the correct way of doing this down? cheers all!!

**Comment by ganigeorgiev** [comment, marked as answer]
Both approaches (when there is a  `customer` relation in each email, address, etc. collection OR when the customer has multiple relations to the other collections as in the screenshot) are fine.

As it is right now (aka. with the multiple `relation` fields) it allows also a single email, address, number, etc. to be shared between multiple customers. If you want to disallow this, you can reverse the relation.

For both cases you can use `expand` (with the `multiple` relation is slightly easier since it is a direct relation; for the inverse relation, the expand expression will be something like `?expand=emails(customer)`).

Note that currently the Web APIs don't support bulk/transaction operations (_it will be added in v0.18/v0.19_), so for both cases you'll have to send individual requests for each record.

When using the `multiple` relation fields, you'll have to create first each email, address, number, etc. records you want to associate to get their id(s) and then create/update the customer.

When the relation is the other way, you have to create first the customer and then use its id to associate it while creating each email, address, number records.

**Reply by ganigeorgiev** [reply]
Also I forgot to mention that if you reverse the relation (_aka. have a single `customer` relation in each email, address, number collection_), when deleting a customer the related entries can be also deleted automatically using the "Cascade delete" option of the relation field.

**Reply by steve-potts** [reply]
thanks again legend :D
